### DISCUSSION TITLE
How to pass and populate field from select field

### DISCUSSION THREAD

**Discussion by MarJose123** [filamentphp/filament#3390] [Help] [answered]
I'm using this code for my form and i want to pass the `id` value to other field in the form.
```php
Select::make('authorId')
    ->label('Author')
    ->options(User::all()->pluck('name', 'id'))
    ->searchable()
```

**Comment by Z3d0X** [comment, marked as answer]
This may be what you are looking for: https://filamentphp.com/docs/2.x/forms/advanced#dependant-fields--components

What are you trying to do?
Elaborating further would allow us to suggest a more suitable solution.

PS: 
`User::pluck('name', 'id')` would be more efficient than `User::all()->pluck('name', 'id')`

**Reply by MarJose123** [reply]
I want to some of the field using the customer information like, once I select the `Author`, then the `Author Number` under the customer form will automatically populated based on the selected author.

**Reply by Z3d0X** [reply]
```php
Select::make('authorId')
    ->label('Author')
    ->options(User::pluck('name', 'id'))
    ->searchable()
    ->reactive()
    ->afterStateUpdated(function ($state, Closure $set) {
        if(blank($state)) return;
        
        $author = Author::find($state);
        
        // $set('author_number', $author->number);
    }),
 ```

**Reply by MarJose123** [reply]
> ```
> Select::make('authorId')
>     ->label('Author')
>     ->options(User::pluck('name', 'id'))
>     ->searchable()
>     ->reactive()
>     ->afterStateUpdated(function ($state, Closure $set) {
>         if(blank($state)) return;
>         
>         $author = Author::find($state);
>         
>         // $set('author_number', $author->number);
>     }),
> ```


Thanks, this is the answer.
